### ISSUE TITLE
Completing a todo might clash with todo.txt rule

### ISSUE THREAD

**Issue by ransome1** [jmhobbs/jsTodoTxt#50]
When you set an item **without a creation date** to complete by adding a completion date (e.g. `item.setCompleted(new Date)`), next branch throws this error: `Error: Can not set completed date without a created date set.`

I understand why this is being done, but if you look at [this rule](https://github.com/todotxt/todo.txt#complete-tasks-2-format-rules), it says that this can be a completed task: `x 2011-03-03 Call Mom`.

jstodotxt does not allow this to happen.

**Comment by jmhobbs** [comment]
You are correct, that is what the spec says.  I must have taken that off the (apparently incorrect) diagram at the top;

<img width="221" alt="Image" src="https://github.com/user-attachments/assets/7d0718cf-a8ba-46cc-adcc-5474836309c7" />

I'll check on #56 and merge it if it's good to go 👍

**Comment by jmhobbs** [comment]
I will make a release after #51 gets resolved 👍
